src/renderer/dx/DxRenderer.cpp

```diff
@@ -1987,7 +1987,18 @@ CATCH_RETURN()
 // - S_OK or relevant DirectX error
 [[nodiscard]] HRESULT DxEngine::UpdateFont(const FontInfoDesired& pfiFontInfoDesired, FontInfo& fiFontInfo) noexcept
 {
-    return UpdateFont(pfiFontInfoDesired, fiFontInfo, {}, {});
+    try
+    {
+        // If no font features or axes were supplied, we call the version of UpdateFont that
+        // tells the DxFontRenderData to continue using the features/axes it already has (if any)
+        RETURN_IF_FAILED(_fontRenderData->UpdateFont(pfiFontInfoDesired, fiFontInfo, _dpi));
+
+        // Prepare the text layout.
+        _customLayout = WRL::Make<CustomTextLayout>(_fontRenderData.get());
+
+        return S_OK;
+    }
+    CATCH_RETURN();

```

Comment:
I also don't like that this is almost exactly the same as the other `UpdateFont` in this file either